src/intersections/lines.jl

```diff
@@ -9,20 +9,24 @@ The intersection type can be one of three types:
 2. overlap at more than one point
 3. do not overlap nor intersect
 =#
-function intersection(f, l1::Line{3,T}, l2::Line{3,T}) where {T}
+function intersection(f::Function, l1::Line{3,T}, l2::Line{3,T}) where {T}
   a, b = l1(0), l1(1)
   c, d = l2(0), l2(1)
 
-  if measure(Tetrahedron(a, b, c, d)) > 0
+  if measure(Tetrahedron(a, b, c, d)) > 0 # not in same plane
     return @IT NoIntersection nothing f
-  elseif isapprox(abs((b - a) × (c - d)), zero(T), atol=atol(T)^2)
-    return @IT OverlappingLines l1 f
+  elseif isapprox(norm((b - a) × (c - d)), zero(T), atol=atol(T)^2)
+    if a in l2
```

Comment:
at this point we know the two lines are in the same plane and parallel,
now we have to check if the lines are collinear

Reply:
I am surprised that the algorithm was missing important branches. Let's wait for the tests to finish running to see if all previous tests are still passing.

Reply:
Add the test `intersect(Line((1,0,0), (0,0,0)), Line((1,1,0), (0,1,0)))`